cluster-api/src/test/java/io/aiven/klaw/clusterapi/services/ApacheKafkaTopicServiceTest.java

```diff
@@ -0,0 +1,461 @@
+  }
+
+  @Test
+  void loadTopicsClientNull() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(null);
+
+    AbstractThrowableAssert<?, ? extends Throwable> exception =
+        assertThatThrownBy(
+            () -> apacheKafkaTopicService.loadTopics(environment, protocol, clusterIdentification));
+
+    exception.isInstanceOf(Exception.class);
+    exception.hasMessage("Cannot connect to cluster.");
+  }
+
+  @Test
+  void loadTopics() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+    Map<String, TopicDescription> topicDescriptionsPerAdminClient =
+        new HashMap<>(
+            Map.of(
+                "_confluent", topicDescription,
+                "__connect", topicDescription,
+                "_schemas", topicDescription,
+                "topic", topicDescription));
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(adminClient);
+    Mockito.when(adminClient.listTopics(any(ListTopicsOptions.class))).thenReturn(listTopicsResult);
+    Mockito.when(listTopicsResult.names()).thenReturn(KafkaFuture.completedFuture(Set.of("name")));
+    Mockito.when(adminClient.describeTopics(any(Collection.class)))
+        .thenReturn(describeTopicsResult);
+    Mockito.when(describeTopicsResult.all())
+        .thenReturn(KafkaFuture.completedFuture(topicDescriptionsPerAdminClient));
+    Mockito.when(topicDescription.partitions()).thenReturn(List.of(topicPartitionInfo));
+
+    Set<TopicConfig> topicConfigs =
+        apacheKafkaTopicService.loadTopics(environment, protocol, clusterIdentification);
+
+    Assertions.assertThat(topicConfigs.size()).isEqualTo(1);
+  }
+
+  @Test
+  void createTopicClientNull() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+    ClusterTopicRequest clusterTopicRequest =
+        ClusterTopicRequest.builder()
+            .env(environment)
+            .clusterName(clusterIdentification)
+            .protocol(protocol)
+            .build();
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(null);
+
+    AbstractThrowableAssert<?, ? extends Throwable> exception =
+        assertThatThrownBy(() -> apacheKafkaTopicService.createTopic(clusterTopicRequest));
+
+    exception.isInstanceOf(Exception.class);
+    exception.hasMessage("Cannot connect to cluster.");
+  }
+
+  @Test
+  void createTopic() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+    String topicName = "TOPIC_NAME";
+    int partitions = 5;
```

Comment:
The following can be made class constants so they only need to be referenced once.
    String environment = "ENVIRONMENT";
    String clusterIdentification = "CLUSTER_IDENTIFICATION";
    String topicName = "TOPIC_NAME";

Reply:
I am making a TestConstants class for this.